src/task/task.service.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  BadRequestException,
+  HttpCode,
+  HttpStatus,
+  Injectable,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Task } from 'src/entities/task.entity';
+import { formatDate } from 'src/helpers/date-and-time.helper';
+import { Repository } from 'typeorm';
+import { UpdateTaskDto } from './dto/update-task.dto';
+import { GetUser } from 'src/decorator/get-user.decorator';
+import { User } from 'src/entities/user.entity';
+
+@Injectable()
+export class TaskService {
+  constructor(
+    @InjectRepository(Task)
+    private taskRepository: Repository<Task>,
+  ) {}
+
+  async getDefaultTaskList(
+    workspaceId: number,
+    userId: number,
+    isForCurrentUserOnly = false,
+  ): Promise<{ tasks: Task[]; count: number }> {
+    const qb = this.taskRepository
+      .createQueryBuilder('tasks')
+      .leftJoinAndSelect('tasks.user', 'user')
+      .select('*')
+      .where('tasks.createdAt like :currentDate', {
+        currentDate: formatDate() + '%',
+      })
+      .andWhere('tasks.workspace = :workspaceId', { workspaceId });
+    if (isForCurrentUserOnly) {
+      qb.andWhere('tasks.user = :userId', { userId });
+    }
+
+    const [tasks, count] = await qb.getManyAndCount();
+    return { tasks, count };
+  }
+
+  async getSingleTask(userId: number): Promise<Task> {
```

Comment:
This is wrong - fetching task by id where it is equal to the user id - it will not work. It should be task id. But generally, I think we'll not need this functionality at all - the list will always be shown.